### PR TITLE
New examples

### PR DESCRIPTION
**Pull request by nstauff**
- Added examples 4 with different optimization methods (scipy.minimization and pymoo) and with one LHS sampling method
- Added link to documentation in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,5 @@ Then you can view the documentation with:
 - google-chrome build/html/index.html
 
 or replace google-chrome with your favorite browser.
+
+Otherwise, here is the link to the online [__documentation__](https://watts.readthedocs.io/en/latest/user/index.html).
```
Comment (paulromano) [inline README.md]: Now that we have online documentation, I think most people won't need to build the documentation locally, so perhaps we should just change this remove the instructions for building locally.

File: examples/example4_main/example4.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 from math import cos, pi
 import os
 import watts
-from scipy.optimize import minimize
-from statistics import mean
+from statistics import mean 
+from statistics import stdev
```
Comment (paulromano) [inline examples/example4_main/example4.py]:
```suggestion
from statistics import mean, stdev
```

File: examples/example4a_opt_scipy/example4a.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# SPDX-License-Identifier: MIT
+
+from scipy.optimize import minimize
+from example4 import *
+
+# initial X values
+X = [0.9, 0.6]
+
+def fitness_calc(X): # simple fitness function to get 1 "objective function" to scipy
+    (keff, max_Tf, avg_Tf) = calc_workflow(X)
```
Comment (paulromano) [inline examples/example4a_opt_scipy/example4a.py]:
```suggestion
def fitness_calc(X):
    """simple fitness function to get 1 "objective function" to scipy"""
    (keff, max_Tf, avg_Tf) = calc_workflow(X)
```

File: examples/example4a_opt_scipy/example4a.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    fitness = abs(keff - 1) + (max_Tf/avg_Tf)
+    return keff
```
Comment (paulromano) [inline examples/example4a_opt_scipy/example4a.py]: Should this be
```suggestion
    return fitness
```

File: examples/example4b_LSH_scipy/example4b.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+from scipy.stats import qmc
+from example4 import *
+
+sampler = qmc.LatinHypercube(d=2)
+sample = sampler.random(n=10)
+l_bounds = [0.5, 0.5]
+u_bounds = [1.0, 1.0]
+sample_scaled = qmc.scale(sample, l_bounds, u_bounds)
+
+print (sample_scaled)
```
Comment (paulromano) [inline examples/example4b_LSH_scipy/example4b.py]:
```suggestion
print(sample_scaled)
```

File: examples/example4b_LSH_scipy/example4b.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+results = {}
+results["keff"] = []
+results["max_Tf"] = []
+results["avg_Tf"] = []
```
Comment (paulromano) [inline examples/example4b_LSH_scipy/example4b.py]:
```suggestion
results = {"keff": [], "max_Tf": [], "avg_Tf": []}
```

File: examples/example4b_LSH_scipy/example4b.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+for X in sample_scaled:
+    res = calc_workflow(X)
+    results["keff"].append(res[0])
+    results["max_Tf"].append(res[1])
+    results["avg_Tf"].append(res[2])
```
Comment (paulromano) [inline examples/example4b_LSH_scipy/example4b.py]:
```suggestion
    keff, max_Tf, avg_Tf = calc_workflow(X)
    results["keff"].append(keff)
    results["max_Tf"].append(max_Tf)
    results["avg_Tf"].append(avg_Tf)
```

File: examples/example4b_LSH_scipy/example4b.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+print (results)
+print ("keff: mean = %f , stdev = %f"%(mean(results["keff"]), stdev(results["keff"])))
+print ("max_Tf: mean = %f , stdev = %f"%(mean(results["max_Tf"]), stdev(results["max_Tf"])))
+print ("avg_Tf: mean = %f , stdev = %f"%(mean(results["avg_Tf"]), stdev(results["avg_Tf"])))
```
Comment (paulromano) [inline examples/example4b_LSH_scipy/example4b.py]:
```suggestion
print(results)
print("keff: {} +/- {}".format(mean(results["keff"]), stdev(results["keff"])))
print("max_Tf: {} +/- {}".format(mean(results["max_Tf"]), stdev(results["max_Tf"])))
print("avg_Tf: {} +/- {}".format(mean(results["avg_Tf"]), stdev(results["avg_Tf"])))
```

File: examples/example4c_opt_pymoo/example4c.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+# SPDX-License-Identifier: MIT
+
+from example4 import *
+import numpy as np
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.optimize import minimize
+from pymoo.problems.functional import FunctionalProblem
+from pymoo.core.problem import Problem
+
+# initial X values
```
Comment (paulromano) [inline examples/example4c_opt_pymoo/example4c.py]: Should this be deleted?

File: examples/example4c_opt_pymoo/example4c.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: 2022 UChicago Argonne, LLC
+# SPDX-License-Identifier: MIT
+
+from example4 import *
+import numpy as np
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.optimize import minimize
+from pymoo.problems.functional import FunctionalProblem
```
Comment (paulromano) [inline examples/example4c_opt_pymoo/example4c.py]:
```suggestion
```

File: examples/example4c_opt_pymoo/example4c.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+# initial X values
+
+# Problem definition
+class fitness_calc(Problem):
+    def __init__(self):
+        super().__init__(n_var=2, n_obj=2, xl=np.array([0.5, 0.5]), xu=np.array([1.0, 1.0]))
+
+    def _evaluate(self, x, out, *args, **kwargs):
+        print (x)
```
Comment (paulromano) [inline examples/example4c_opt_pymoo/example4c.py]:
```suggestion
        print(x)
```

File: examples/example4c_opt_pymoo/example4c.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        out["F"] = [] 
+        for X in x:
```
Comment (paulromano) [inline examples/example4c_opt_pymoo/example4c.py]: I generally wouldn't use variable names that only differ by capitalization (`X` and `x`) -- can you change the `X` to something else?

File: examples/example4c_opt_pymoo/example4c.py [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+            (keff, max_Tf, avg_Tf) = calc_workflow(X)
+            print (keff, max_Tf, avg_Tf)
```
Comment (paulromano) [inline examples/example4c_opt_pymoo/example4c.py]:
```suggestion
            print(keff, max_Tf, avg_Tf) 
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for the new examples -- cool to see some different optimization packages in use!

File: README.md (hunk 1 above)
Comment (nstauff) [reply README.md]: Right I changed the order to provide first the link, then the option to generate the documentation locally.

**Comment by nstauff** [comment]
@paulromano I just made all the changes suggested. Thanks for the detailed review! Feel free to merge the PR if you are happy with it.

**Review by paulromano — APPROVED** [review]

**Comment by paulromano** [comment]
Thanks @nstauff!

**Pull request merged**